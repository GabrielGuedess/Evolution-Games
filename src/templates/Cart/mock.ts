import { cartItems } from 'components/organisms/Navbar/mock';

export default cartItems;
