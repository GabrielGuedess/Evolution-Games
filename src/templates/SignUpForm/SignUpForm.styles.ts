import media from 'styled-media-query';

import styled, { css } from 'styled-components';

type CircleIndicatorsProps = {
  isActive: boolean;
  allowClick: boolean;
};

export const Wrapper = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    padding: 5rem 0;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 25rem;
    z-index: ${theme.layers.base};

    > button {
      max-width: 39.4rem;

      span {
        padding: 0.9rem 0;
      }
    }
  `}
`;

export const Header = styled.header`
  ${({ theme }) => css`
    position: absolute;
    inset: 4rem 0 auto 0;
    z-index: ${theme.layers.base};
  `}
`;

export const Relative = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 5rem;
`;

export const WrapperIndicators = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  gap: 5rem;

  ${media.greaterThan('medium')`
      gap: 6.5rem;
    `}
`;

export const CircleIndicators = styled.div<CircleIndicatorsProps>`
  ${({ theme, isActive, allowClick }) => css`
    width: 4rem;
    aspect-ratio: 1/1;
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    background: ${isActive ? theme.colors.primary : 'transparent'};
    box-shadow: ${isActive && theme.shadows.box};
    border: ${!isActive && `2px solid ${theme.colors.secondary}`};
    cursor: ${allowClick ? 'pointer' : 'not-allowed'};
  `}
`;

export const LineIndicator = styled.div`
  ${({ theme }) => css`
    position: absolute;
    left: 100%;
    top: 50%;
    width: calc(5rem + 0.2rem);
    height: 0.2rem;
    transform: translate(0, -50%);
    background: ${theme.colors.secondary};
    z-index: -1;

    ${media.greaterThan('medium')`
      width: calc(6.5rem + 0.2rem);
    `}
  `}
`;

export const NumberIndicator = styled.div`
  ${({ theme }) => css`
    font-size: 1.5rem;
    text-align: center;
    color: ${theme.colors.white};
  `}
`;

export const TitleIndicator = styled.div`
  width: 6rem;
  text-align: center;
  font-size: 0.8rem;
  position: absolute;
  top: 6.2rem;
  left: 50%;
  transform: translate(-50%, -50%);
`;
