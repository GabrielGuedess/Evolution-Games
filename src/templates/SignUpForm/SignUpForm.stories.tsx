import { Story, Meta } from '@storybook/react/types-6-0';

import { steps } from './mock';
import { SignUpForm, SignUpFormProps } from './SignUpForm';

export default {
  title: 'Templates/SignUpForm',
  component: SignUpForm,
  args: {
    steps,
  },
} as Meta<SignUpFormProps>;

export const Default: Story<SignUpFormProps> = args => <SignUpForm {...args} />;
