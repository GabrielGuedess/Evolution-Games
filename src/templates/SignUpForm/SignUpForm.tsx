import { useCallback, useState } from 'react';
import Particles from 'react-particles';

import { Check, CircleNotch } from 'phosphor-react';
import type { Engine, RecursivePartial, IOptions } from 'tsparticles-engine';
import { loadFireworksPreset } from 'tsparticles-preset-fireworks';
import requestFake from 'utils/requestFake';

import Button from 'components/atoms/Button/Button';
import { Logo } from 'components/atoms/Logo/Logo';
import { SignUpData } from 'components/organisms/SignUpData/SignUpData';
import { SignUpFinally } from 'components/organisms/SignUpFinally/SignUpFinally';
import { SignUpLocation } from 'components/organisms/SignUpLocation/SignUpLocation';
import { SignUpUser } from 'components/organisms/SignUpUser/SignUpUser';

import * as S from './SignUpForm.styles';

export type Steps = {
  id: string;
  title: string;
};

export type SignUpFormProps = {
  steps: Steps[];
};

export const SignUpForm = ({ steps }: SignUpFormProps) => {
  const [currentStep, setCurrentStep] = useState(0);
  const [loading, setLoading] = useState(false);

  const [userPhoto, setUserPhoto] = useState<File | null>(null);

  const [confirm, setConfirm] = useState(false);

  const [userInputs, setUserInputs] = useState({
    email: '',
    username: '',
    password: '',
    confirmPassword: '',
  });

  const [dataInputs, setDataInputs] = useState({
    name: '',
    lastName: '',
    cpf: '',
    cellphone: '',
    date: '',
  });

  const [locationInputs, setLocationInputs] = useState({
    cep: '',
    logradouro: '',
    numero: '',
    complemento: '',
    referencia: '',
    bairro: '',
    cidade: '',
    uf: '',
  });

  const handleSubmit = async () => {
    try {
      setLoading(true);

      await requestFake();

      setConfirm(true);
      setCurrentStep(currentStep + 1);
    } catch (error) {
      setConfirm(false);
    } finally {
      setLoading(false);
    }
  };

  const forms = [
    <SignUpUser
      currentStep={currentStep}
      setCurrentStep={setCurrentStep}
      userPhoto={userPhoto}
      setUserPhoto={setUserPhoto}
      userInputs={userInputs}
      setUserInputs={setUserInputs}
    />,
    <SignUpData
      currentStep={currentStep}
      setCurrentStep={setCurrentStep}
      dataInputs={dataInputs}
      setDataInputs={setDataInputs}
    />,
    <SignUpLocation
      currentStep={currentStep}
      setCurrentStep={setCurrentStep}
      locationInputs={locationInputs}
      setLocationInputs={setLocationInputs}
    />,
    <Button
      disabled={loading}
      size="large"
      fullWidth
      onClick={handleSubmit}
      icon={
        loading ? (
          <CircleNotch size={24} className="animation-spinner">
            <animateTransform
              attributeName="transform"
              attributeType="XML"
              type="rotate"
              dur="0.6s"
              from="0 0 0"
              to="360 0 0"
              repeatCount="indefinite"
            />
          </CircleNotch>
        ) : undefined
      }
    >
      Confirmar
    </Button>,
    <SignUpFinally name={dataInputs.name} />,
  ];

  const particlesInit = useCallback(async (engine: Engine) => {
    await loadFireworksPreset(engine);
  }, []);

  const options: RecursivePartial<IOptions> = {
    background: {
      image: 'url(/img/sign-in-background-image.png)',
    },
    particles: {
      move: {
        trail: {
          fillColor: '#0000000',
        },
      },
    },
    preset: 'fireworks',
  };

  return (
    <S.Wrapper>
      <S.Header>
        <S.Relative>
          <Logo />

          <S.WrapperIndicators>
            {steps.map((step, i, row) => (
              <S.CircleIndicators
                key={step.id}
                isActive={currentStep >= i}
                allowClick={!confirm && currentStep > i}
                onClick={() => !confirm && currentStep > i && setCurrentStep(i)}
              >
                {currentStep > i ? (
                  <Check size={18} weight="bold" />
                ) : (
                  <S.NumberIndicator>{i + 1}</S.NumberIndicator>
                )}
                {i + 1 !== row.length && <S.LineIndicator />}

                <S.TitleIndicator>{step.title}</S.TitleIndicator>
              </S.CircleIndicators>
            ))}
          </S.WrapperIndicators>
        </S.Relative>
      </S.Header>

      <S.Content>{forms[currentStep]}</S.Content>

      {forms.length - 1 === currentStep && (
        <Particles
          init={particlesInit}
          options={options}
          style={{ zIndex: -1 }}
        />
      )}
    </S.Wrapper>
  );
};
