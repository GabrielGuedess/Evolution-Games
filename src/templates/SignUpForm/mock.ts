import { Steps } from './SignUpForm';

export const steps: Steps[] = [
  {
    id: 'user',
    title: 'Usuário',
  },
  {
    id: 'data',
    title: 'Dados pessoais',
  },
  {
    id: 'local',
    title: 'Localização',
  },
  {
    id: 'finally',
    title: 'Finalização',
  },
];
