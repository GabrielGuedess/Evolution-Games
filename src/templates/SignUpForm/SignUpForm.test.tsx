import { renderWithProviders } from 'utils/tests/helpers';

import { steps } from './mock';
import { SignUpForm } from './SignUpForm';

describe('<SignUpForm />', () => {
  it('should render the heading', () => {
    const { container } = renderWithProviders(<SignUpForm steps={steps} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
