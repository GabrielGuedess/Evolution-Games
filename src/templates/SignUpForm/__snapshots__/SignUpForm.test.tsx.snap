// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<SignUpForm /> should render the heading 1'] = `
.c23 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  position: relative;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 5px;
  -webkit-transition: all 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: all 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: #7C3AED;
  color: #F4F4F5;
  width: 100%;
}

.c23:not(:disabled):is(:hover,:focus) {
  box-shadow: 0 0.8rem 2.8rem rgba(118,74,241,0.5);
  color: #F4F4F5;
}

.c23 .effect {
  fill: #7C3AED;
}

.c23:disabled {
  cursor: not-allowed;
  -webkit-filter: saturate(30%);
  filter: saturate(30%);
}

.c23:not(:disabled):is(:hover,:focus) {
  -webkit-filter: brightness(1.1);
  filter: brightness(1.1);
}

.c23:not(:disabled):active {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c23 span {
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 1;
  display: block;
  position: relative;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  -webkit-transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  z-index: 1;
}

.c23:not(:disabled):is(:hover,:focus) span {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c23:not(:disabled):is(:hover,:focus) .icon {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c23:not(:disabled):active span {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c23 svg {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  -webkit-transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: transform 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
}

.c23 .effect {
  position: absolute;
  top: initial;
  left: initial;
  width: 100%;
  height: 100%;
  z-index: -1;
}

.c23 .effect path {
  -webkit-transition: 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
  transition: 0.3s cubic-bezier(0.68,-0.6,0.32,1.6);
}

.c23:not(:disabled):is(:hover,:focus) .effect path {
  d: path( 'M0,0 C0,-5 100,-5 100,0 C105,0 105,100 100,100 C100,105 0,105 0,100 C-5,100 -5,0 0,0' );
}

.c23:not(:disabled):active .effect path {
  d: path( 'M0 5C0 2.23858 2.23858 0 5 0H95C97.7614 0 100 2.23858 100 5V95C100 97.7614 97.7614 100 95 100H5C2.23858 100 0 97.7614 0 95V5Z' );
}

.c3 {
  width: 8.6rem;
  height: 2rem;
  color: #FFFFFF;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c16 {
  position: relative;
}

.c19 {
  position: absolute;
  left: 1rem;
  top: 1.7rem;
  padding: 0 3px;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  z-index: 2;
}

.c19::before {
  content: '';
  position: absolute;
  inset: 0;
  width: 100%;
  z-index: -1;
}

.c17 {
  width: 100%;
  font-size: 1.8rem;
  font-weight: 400;
  font-style: normal;
  line-height: 3.2rem;
  border-width: 2px;
  border-style: solid;
  border-color: #272530;
  background-color: transparent;
  border-radius: 0.5rem;
  padding: 0.8rem;
  color: #FFFFFF;
  outline: none;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c17::-webkit-calendar-picker-indicator {
  background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="15" viewBox="0 0 24 24"><path fill="%23272530" d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"/></svg>');
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c17 .c18 {
  color: #FFFFFF;
}

.c17:-webkit-autofill {
  -webkit-box-shadow: 0 0 0px 100rem #04040C inset;
  -webkit-text-fill-color: #FFFFFF;
}

.c17:-webkit-autofill ~ div,
.c17:-webkit-autofill:hover ~ div,
.c17:-webkit-autofill:focus ~ div,
.c17:-webkit-autofill:active ~ div {
  color: #717B8A;
}

.c17::-webkit-input-placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c17::-moz-placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c17:-ms-input-placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c17::placeholder {
  font-size: 1.4rem;
  opacity: 0;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c17:focus {
  border-color: #7C3AED;
}

.c17:focus::-webkit-calendar-picker-indicator {
  background-image: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="15" viewBox="0 0 24 24"><path fill="%237C3AED" d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"/></svg>');
}

.c17:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c17:focus::-moz-placeholder {
  opacity: 1;
}

.c17:focus:-ms-input-placeholder {
  opacity: 1;
}

.c17:focus::placeholder {
  opacity: 1;
}

.c17:is(:focus,:valid) + .c18 {
  top: -2rem;
  color: #7C3AED;
  font-size: 1.2rem;
}

.c17::selection {
  color: #FFFFFF;
  background: #7C3AED;
}

.c20 {
  width: 50px;
  height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  right: 0;
  top: 0;
}

.c11 {
  width: 100%;
  max-width: 40.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3.2rem;
}

.c11 .animation-spinner {
  -webkit-transition: none;
  transition: none;
}

.c11 > button span {
  padding: 0.9rem 0;
}

.c12 {
  position: relative;
  width: 25rem;
  aspect-ratio: 1/1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  border: 2px solid #272530;
  margin-bottom: 3.2rem;
  cursor: pointer;
  -webkit-transition: 0.3s all;
  transition: 0.3s all;
}

.c12:is(:hover,:focus) {
  border: 2px solid #7C3AED;
}

.c12:is(:hover,:focus) svg {
  color: #7C3AED;
}

.c13 {
  color: #717B8A;
  position: absolute;
  inset: auto 0 0 auto;
  cursor: pointer;
  -webkit-transition: 0.3s all;
  transition: 0.3s all;
  outline: none;
}

.c13:is(:hover,:focus) {
  color: #7C3AED;
}

.c14 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3.2rem;
}

.c21 {
  width: 2rem;
  height: 2rem;
  fill: none;
}

.c21 polyline,
.c21 path {
  stroke-width: 1.6;
  stroke: #FFFFFF;
}

.c21 path {
  stroke-dasharray: 46;
  stroke-dashoffset: 150;
  -webkit-transition: stroke-dasharray 0.6s ease 0s, stroke-dashoffset 0.8s ease 0.3s,stroke 0.3s ease 0.6s;
  transition: stroke-dasharray 0.6s ease 0s, stroke-dashoffset 0.8s ease 0.3s,stroke 0.3s ease 0.6s;
}

.c21 polyline {
  stroke-dashoffset: 0;
  -webkit-transition: stroke-dashoffset 0.5s ease 0.6s,stroke 0.3s ease 0.6s;
  transition: stroke-dashoffset 0.5s ease 0.6s,stroke 0.3s ease 0.6s;
}

.c22 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
}

.c10 {
  position: relative;
  width: 100%;
  padding: 5rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 25rem;
  z-index: 10;
}

.c10 > button {
  max-width: 39.4rem;
}

.c10 > button span {
  padding: 0.9rem 0;
}

.c1 {
  position: absolute;
  inset: 4rem 0 auto 0;
  z-index: 10;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5rem;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 5rem;
}

.c5 {
  width: 4rem;
  aspect-ratio: 1/1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  background: #7C3AED;
  box-shadow: 0 0.8rem 2.8rem rgba(118,74,241,0.5);
  cursor: not-allowed;
}

.c9 {
  width: 4rem;
  aspect-ratio: 1/1;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  background: transparent;
  border: 2px solid #717B8A;
  cursor: not-allowed;
}

.c7 {
  position: absolute;
  left: 100%;
  top: 50%;
  width: calc(5rem + 0.2rem);
  height: 0.2rem;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  background: #717B8A;
  z-index: -1;
}

.c6 {
  font-size: 1.5rem;
  text-align: center;
  color: #FFFFFF;
}

.c8 {
  width: 6rem;
  text-align: center;
  font-size: 0.8rem;
  position: absolute;
  top: 6.2rem;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

@media (min-width:450px) {
  .c23 .effect {
    top: -5%;
    left: -5%;
    width: 110%;
    height: 110%;
  }
}

@media (min-width:1170px) {
  .c3 {
    width: 13.7rem;
    height: 3.2rem;
    color: #7C3AED;
  }
}

@media (min-width:768px) {
  .c19::before {
    background: #04040C;
  }
}

@media (min-width:768px) {
  .c17:is(:focus,:valid) + .c18 {
    top: -8px;
  }
}

@media (min-width:768px) {
  .c14 {
    gap: 1.6rem;
  }
}

@media (min-width:768px) {
  .c4 {
    gap: 6.5rem;
  }
}

@media (min-width:768px) {
  .c7 {
    width: calc(6.5rem + 0.2rem);
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-label="Evolution"
          fill="none"
          role="img"
          viewBox="0 0 134 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.344 18.104V23H0.504V0.919998H16.344V5.816H6.552V9.496H16.28V14.392H6.552V18.104H16.344ZM28.9195 23H24.2795L17.9115 7H24.0875L26.6795 15.704L29.1115 7H35.0955L28.9195 23ZM41.9485 18.36C42.2898 18.936 42.8872 19.224 43.7405 19.224C44.5938 19.224 45.1912 18.936 45.5325 18.36C45.8952 17.784 46.0765 16.696 46.0765 15.096C46.0765 13.4107 45.8952 12.2693 45.5325 11.672C45.1912 11.0747 44.5938 10.776 43.7405 10.776C42.8872 10.776 42.2898 11.0747 41.9485 11.672C41.6072 12.2693 41.4365 13.4107 41.4365 15.096C41.4365 16.696 41.6072 17.784 41.9485 18.36ZM43.7405 6.584C46.5992 6.584 48.6258 7.20267 49.8205 8.44C51.0152 9.656 51.6125 11.8747 51.6125 15.096C51.6125 18.232 51.0045 20.408 49.7885 21.624C48.5938 22.8187 46.5778 23.416 43.7405 23.416C40.9245 23.416 38.9192 22.8187 37.7245 21.624C36.5298 20.408 35.9325 18.232 35.9325 15.096C35.9325 11.8747 36.5298 9.656 37.7245 8.44C38.9192 7.20267 40.9245 6.584 43.7405 6.584Z"
            fill="currentColor"
          />
          <path
            d="M59.5995 0.919998V23H54.0635V0.919998H59.5995ZM67.6888 7V17.464C67.6888 18.1253 67.8381 18.584 68.1368 18.84C68.4568 19.096 68.9368 19.224 69.5768 19.224C70.9208 19.224 71.5928 18.4987 71.5928 17.048V7H77.1287V23H71.5928V21.112H71.0488C70.1314 22.648 68.7981 23.416 67.0488 23.416C65.5768 23.416 64.3928 23.0427 63.4968 22.296C62.6008 21.528 62.1528 20.28 62.1528 18.552V7H67.6888ZM90.5328 18.392V23H86.8208C82.9808 23 81.0608 21.1333 81.0608 17.4V11.32H78.8528V7H81.0608V2.2H86.5968V7H90.5328V11.32H86.5968V16.856C86.5968 17.4107 86.7248 17.8053 86.9808 18.04C87.2368 18.2747 87.6634 18.392 88.2608 18.392H90.5328ZM92.6908 7H98.1948V23H92.6908V7ZM98.3228 5.016H92.5628V0.952H98.3228V5.016ZM106.792 18.36C107.134 18.936 107.731 19.224 108.584 19.224C109.438 19.224 110.035 18.936 110.376 18.36C110.739 17.784 110.92 16.696 110.92 15.096C110.92 13.4107 110.739 12.2693 110.376 11.672C110.035 11.0747 109.438 10.776 108.584 10.776C107.731 10.776 107.134 11.0747 106.792 11.672C106.451 12.2693 106.28 13.4107 106.28 15.096C106.28 16.696 106.451 17.784 106.792 18.36ZM108.584 6.584C111.443 6.584 113.47 7.20267 114.664 8.44C115.859 9.656 116.456 11.8747 116.456 15.096C116.456 18.232 115.848 20.408 114.632 21.624C113.438 22.8187 111.422 23.416 108.584 23.416C105.768 23.416 103.763 22.8187 102.568 21.624C101.374 20.408 100.776 18.232 100.776 15.096C100.776 11.8747 101.374 9.656 102.568 8.44C103.763 7.20267 105.768 6.584 108.584 6.584ZM128.315 23V12.536C128.315 11.8747 128.155 11.416 127.835 11.16C127.537 10.904 127.057 10.776 126.395 10.776C125.051 10.776 124.379 11.5013 124.379 12.952V23H118.875V7H124.379V8.888H124.923C125.819 7.352 127.217 6.584 129.115 6.584C130.651 6.584 131.814 6.95733 132.603 7.704C133.414 8.45067 133.819 9.69867 133.819 11.448V23H128.315Z"
            fill="white"
          />
        </svg>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            1
          </div>
          <div
            class="c7"
          />
          <div
            class="c8"
          >
            Usuário
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c6"
          >
            2
          </div>
          <div
            class="c7"
          />
          <div
            class="c8"
          >
            Dados pessoais
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c6"
          >
            3
          </div>
          <div
            class="c7"
          />
          <div
            class="c8"
          >
            Localização
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c6"
          >
            4
          </div>
          <div
            class="c8"
          >
            Finalização
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="c10"
  >
    <form
      class="c11"
      role="form"
    >
      <input
        accept="image/*"
        style="display: none;"
        type="file"
      />
      <div
        class="c12"
      >
        <div
          class="go2017755475"
          color="ECFAFE"
          size="240"
        >
          <p
            class="go3451136944"
            color="1F84A3"
            size="240"
          />
        </div>
        <button
          aria-label="User Photo"
          class="c13"
          type="button"
        >
          <svg
            fill="currentColor"
            height="30"
            viewBox="0 0 256 256"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <path
              d="M208,208H48a16,16,0,0,1-16-16V80A16,16,0,0,1,48,64H80L96,40h64l16,24h32a16,16,0,0,1,16,16V192A16,16,0,0,1,208,208Z"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
            <circle
              cx="128"
              cy="132"
              fill="none"
              r="36"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </button>
      </div>
      <div
        class="c14"
      >
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <input
              autocomplete="on"
              class="c17"
              id="email"
              name="email"
              placeholder="Insira seu e-mail"
              required=""
              type="email"
              value=""
            />
            <label
              aria-label="email"
              class="c18 c19"
              for="email"
            >
              E-mail
            </label>
            <div
              class="c20"
            >
              <svg
                class="c21"
                viewBox="0 0 16 16"
              >
                <path
                  d="M10.8000002,10.8000002 C9.85000038,11.6500006 9.18349609,12 8,12 C5.80000019,12 4,10.1999998 4,8 C4,5.80000019 5.80000019,4 8,4 C10.1999998,4 12,6 12,8 C12,9.35332031 12.75,9.5 13.5,9.5 C14.25,9.5 15,8.60000038 15,8 C15,4 12,1 8,1 C4,1 1,4 1,8 C1,12 4,15 8,15 C12,15 15,12 15,8"
                />
                <polyline
                  points="5 8.375 7.59090909 11 14.5 4"
                  transform="translate(0 -0.5)"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <input
              autocomplete="on"
              class="c17"
              id="username"
              name="username"
              placeholder="Insira sua senha"
              required=""
              value=""
            />
            <label
              aria-label="username"
              class="c18 c19"
              for="username"
            >
              Username
            </label>
          </div>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <input
              autocomplete="on"
              class="c17"
              id="password"
              name="password"
              placeholder="Insira sua senha"
              required=""
              type="password"
              value=""
            />
            <label
              aria-label="password"
              class="c18 c19"
              for="password"
            >
              Senha
            </label>
            <div
              class="c20"
            >
              <div
                class="c22"
              >
                <svg
                  aria-label="Mostrar senha"
                  fill="currentColor"
                  height="26"
                  viewBox="0 0 256 256"
                  width="26"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="256"
                    width="256"
                  />
                  <path
                    d="M128,56C48,56,16,128,16,128s32,72,112,72,112-72,112-72S208,56,128,56Z"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                  />
                  <circle
                    cx="128"
                    cy="128"
                    fill="none"
                    r="40"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="16"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <input
              autocomplete="on"
              class="c17"
              id="confirm-password"
              name="confirmPassword"
              placeholder="Confirme sua senha"
              required=""
              type="password"
              value=""
            />
            <label
              aria-label="confirm-password"
              class="c18 c19"
              for="confirm-password"
            >
              Confirmar Senha
            </label>
          </div>
        </div>
      </div>
      <button
        aria-label="Button Action"
        class="c23"
        disabled=""
        type="submit"
      >
        <span>
          Próximo
        </span>
        <svg
          aria-hidden="true"
          class="effect"
          preserveAspectRatio="none"
          viewBox="-5 -5 110 110"
        >
          <path
            d="M0 8C0 3.58172 3.58172 0 8 0H92C96.4183 0 100 3.58172 100 8V92C100 96.4183 96.4183 100 92 100H8C3.58172 100 0 96.4183 0 92V8Z"
          />
        </svg>
      </button>
    </form>
  </div>
</div>
`;
